### PR TITLE
JSAB2-50 Battle map backend

### PR DESCRIPTION
**Pull request by GuJialu**

### CODE REVIEW

**Review by HONGTAO-T — APPROVED** [review]

**Review by HanqingZ — APPROVED** [review]

**Review by kyya — APPROVED** [review]

**Review by HanqingZ — APPROVED** [review]

**Review by HONGTAO-T — APPROVED** [review]

File: backend/routers/battle.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const { Router } = require('express');
+const { UserModel } = require('../models');
+const { TroopModel } = require('../models');
+const { auth } = require('../middlewares');
+
+const router = Router();
+
+async function getUsersInPlanet(req, res, next) {
```
Comment (evasimonyi) [inline backend/routers/battle.js]:
```suggestion
async function getUsersOnPlanet(req, res, next) {
```

File: backend/routers/battle.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const { _id: userId } = req.user;
+  const { planet } = req.params;
+  try {
+    const usersInPlanet = await UserModel.find(
```
Comment (evasimonyi) [inline backend/routers/battle.js]:
```suggestion
    const usersOnPlanet = await UserModel.find(
```

File: backend/routers/battle.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      {
+        planetList: planet,
+        _id: { $ne: userId },
+      },
+      {
+        email: false,
+        password: false,
+      },
+    );
+    const usersInPlanetFiltered = await Promise.all(usersInPlanet.map(async (element) => {
```
Comment (evasimonyi) [inline backend/routers/battle.js]:
```suggestion
    const usersOnPlanetFiltered = await Promise.all(usersOnPlanet.map(async (element) => {
```

File: backend/routers/battle.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        _id,
+        username: element.username,
+        hp: troop ? troop.hp : 0,
+        attack: troop ? troop.attack : 0,
+        defence: troop ? troop.defence : 0,
+        allLevels: troop ? troop.battleRating : 0,
+        planetList: element.planetList,
+      };
+    }));
+    res.send({ usersInPlanet: usersInPlanetFiltered });
```
Comment (evasimonyi) [inline backend/routers/battle.js]:
```suggestion
    res.send({ usersOnPlanet: usersOnPlanetFiltered });
```

File: backend/routers/battle.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  } catch (error) {
+    next(error);
+  }
+}
+
+router.get('/planet/:planet', auth, getUsersInPlanet);
```
Comment (evasimonyi) [inline backend/routers/battle.js]:
```suggestion
router.get('/planet/:planet', auth, getUsersOnPlanet);
```

:planet: name or id?

**Review by evasimonyi — CHANGES_REQUESTED** [review]
please change the naming of `usersInPlanet` to `usersOnPlanet`, cause it's grammatically incorrect:) thank you

very nice job!

File: backend/routers/battle.js (hunk 5 above)
Comment (GuJialu) [reply backend/routers/battle.js]: It's the name but also works as id in our design.

**Review by evasimonyi — APPROVED** [review]

**Pull request merged**